scripts/oneoff/rename-dos-locations.py

```diff
@@ -0,0 +1,67 @@
+"""Rename location fields
+
+Rename location fields for digital-outcomes and user-research-participants.
+
+Usage:
+    scripts/oneoff/rename-dos-locations.py <stage> <token> [--dry-run]
+"""
+import sys
+sys.path.insert(0, '.')
+
+from multiprocessing.pool import ThreadPool
+import itertools
+from docopt import docopt
+from dmapiclient import DataAPIClient
+from dmscripts.env import get_api_endpoint_from_stage
+
+
+def create_draft_getter(client):
+    def get_drafts(supplier):
+        for draft in client.find_draft_services_iter(supplier['id'], framework='digital-outcomes-and-specialists'):
+            yield draft
+    return get_drafts
+
+
+def get_all_drafts(client):
+    pool = ThreadPool(25)
+    return itertools.chain.from_iterable(pool.imap_unordered(
+        create_draft_getter(client),
+        client.find_suppliers_iter()
+    ))
+
+
+if __name__ == "__main__":
+    arguments = docopt(__doc__)
+
+    stage = arguments['<stage>']
+    api_token = arguments['<token>']
+    dry_run = arguments['--dry-run']
+
+    api_url = get_api_endpoint_from_stage(stage)
+    client = DataAPIClient(api_url, api_token)
+
+    counter = 0
+    for draft in get_all_drafts(client):
+        update = {}
+        if "outcomesLocations" in draft:
+            update["locations"] = draft["outcomesLocations"]
+            update["outcomesLocations"] = None
+        if "recruitLocations" in draft:
+            if update:
+                raise ValueError("draft {} has both outcomesLocations and recruitLocations".format(draft["id"]))
+            update["locations"] = draft["recruitLocations"]
+            update["recruitLocations"] = None
+
+        if update:
+            counter += 1
+
+            update_reason = "rename locations: https://www.pivotaltracker.com/story/show/113271913"
+            if dry_run:
+                print("{} {}".format(draft["id"], update))
+            else:
+                client.update_draft_service(draft["id"], update, update_reason)
```

Comment:
I'm still not sure about this. I know the user field is not strictly machine readable since we put both user emails and `whoami` there, but it's still `user`, so I think you could expect to see a name or a reference there. Reason text would look like the fields got mixed up. Putting a script name instead of reason might be a good middle ground.

I've just remembered another problem with draft updates though – [we actually look up the last event user in the API](https://github.com/alphagov/digitalmarketplace-api/blob/master/app/main/views/drafts.py#L171). This is then [used by the supplier frontend to show "Last edited by"](https://github.com/alphagov/digitalmarketplace-supplier-frontend/blob/master/app/templates/services/service_submission.html#L76). So if we do this without setting the user to an existing user email, it will show `Last edited <date> by `. I'm not sure what's the best approach here. We could use the admin account email address to make the message look right, but then we're not actually making any visible changes to the service data.